python-package/xgboost/data.py

```diff
@@ -247,7 +247,25 @@ def _is_modin_df(data: DataType) -> bool:
     "boolean": "i",
 }
 
+pandas_pyarrow_mapper = {
+    "int8[pyarrow]": "i",
+    "int16[pyarrow]": "i",
+    "int32[pyarrow]": "i",
+    "int64[pyarrow]": "i",
+    "uint8[pyarrow]": "i",
+    "uint16[pyarrow]": "i",
+    "uint32[pyarrow]": "i",
+    "uint64[pyarrow]": "i",
+    "float[pyarrow]": "float",
+    "float32[pyarrow]": "float",
+    "double[pyarrow]": "float",
+    "float64[pyarrow]": "float",
+    "bool[pyarrow]": "i",
+}
```

Comment:
I could totally be missing something as I'm not familiar with how xgboost handles data types internally. Does this mapping somehow mean all the pyarrow int types are mapped to the same representation in xgboost? Similar question for float data

Reply:
yes. all of them are treated as float32 inside xgboost. The string "i", and "float" are used for visualization and categorical split (which is not yet available for pyarrow for now due to mentioned error).